esp/templates/survey/review.html

```diff
@@ -12,19 +12,68 @@
 <style type="text/css">
 .nocheckmark { border: 1px solid black; }
 </style>
-<br /><br />
-<h1>Survey Responses for {{ program.niceName }} </h1>
+<a name="top"></a><h1>Survey Responses for {{ program.niceName }} </h1>
+
+{% if teacher_form %}
+<form method="POST" action="{{ request.get_full_path }}">
+<h2>Search for a teacher to retrieve their specific survey results:</h2>
+{{ teacher_form.target_user }}
+<input type="submit" value="Search" />
+</form>
+<br><br>
+{% endif %}
 
 <div id="program_form">
+
+{% if teacher %}
 <center>
+<h2>Survey results for {{ teacher.name }}</h2>
+</center>
+{% endif %}
+
+
+<h2>Table of Contents</h2>
+<ul>
+{% for s in surveys %}
+    <li><a href="#survey{{ s.id }}">Responses for: {{ s.name }}</a>
+    <ul style="list-style-type:circle;">
+    {% for q in s.display_data.questions %}
+        {% ifchanged q.question.per_class %}
+            <li><a href="#survey{{ s.id }}-{{ q.question.per_class }}">{% if q.question.per_class %}Per Class{% else %}Individual{% endif %} Questions</a></li>
+        {% endifchanged %}
+    {% endfor %}
+    </ul>
+    </li>
+{% endfor %}
+
+{% if perclass_data %}
+{% for c in perclass_data %}
+    {% ifchanged c.class.parent_class %}
+        {% if not forloop.first %}
+            </ul>
+        {% endif %}
+        <li><a href="#class{{ c.class.parent_class.id }}">Responses for {{ c.class.parent_class.emailcode }}: {{ c.class.title }}</a></li>
```

Comment:
Hmm for me all the perclass data for a question shows up in one list of responses, not separated by class, even though the code looks like there are class headings? I get the class headings when I'm filtered to a teacher, but not on the overall page.

Reply:
Oh, I hadn't thought about that. Umm... I'll see if it's possible to separate by class for this section.